File: chinese-auction-api/ChineseAuction.Api/Models/Order.cs
namespace ChineseAuction.Api.Models
{
    public enum Status
    {
       IsDraft,
       IsConfirmed
    }

    public class Order
    {
        public int Id { get; set; } // ID הזמנה
        public DateTime OrderDate { get; set; } // תאריך ביצוע ההזמנה
        public Status Status { get; set; } = Status.IsDraft;//  נשמר כטיוטה ורק לאחר אישור נרכש בפועל 
        public decimal TotalAmount { get; set; } // הסכום הכולל של ההזמנה
        public int UserId { get; set; } // קשר לרוכש: מי ביצע את ההזמנה
        public User User { get; set; } = null!;// קשר לרוכש: מי ביצע את ההזמנה

        public ICollection<OrderItem> OrderItems { get; set; } = new List<OrderItem>(); // קשר לפרטי הזמנה – אילו פריטים כלולים בהזמנה זו
    }
}

File: chinese-auction-web/src/app/models/order.ts
import { User } from './user';
import { OrderItem } from './order-item';

export enum Status {
    IsDraft = 'IsDraft',
    IsConfirmed = 'IsConfirmed'
}

export interface Order {
    id: number;
    orderDate: Date;
    status: Status;
    totalAmount: number;
    userId: number;
    user: User;
    orderItems: OrderItem[];
}

export interface OrderItemCreateDto {
  giftId: number;
  quantity: number;
}


export interface OrderItemResponseDto {
    id: number;        // <--- חובה להוסיף את זה! (זה ה-OrderItemId למחיקה)
    giftId: number;
    giftName: string;
    quantity: number;
    unitPrice: number;
}

export interface OrderResponseDto {
    id: number;          // חובה: מזהה ההזמנה
    userId: number;
    orderDate: Date;
    totalAmount: number;
    orderItems: OrderItemResponseDto[];
}

